### PR TITLE
sorter/leveldb(ticdc): fix system stop blocks forever

### LINKED ISSUE #4699: [CC scenario] some captures' sink don't work
### What did you do?

- CDC 0 & 7 sink write are 0 rows/s:
image
- In the same time, we can see all captures are alive:
image

### What did you expect to see?

_No response_

### What did you see instead?

As above.

### Versions of the cluster

Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

```console
v5.4.0
```

Upstream TiKV version (execute `tikv-server --version`):

```console
v5.4.0
```

TiCDC version (execute `cdc version`):

```console
v5.4.0
```

### PR DESCRIPTION
**Pull request by overvenus**
### What problem does this PR solve?

Issue Number: close #4699

Needs to merge #4819 first.

### What is changed and how it works?

StopMessage maybe lost if the total broadcast time exceeds  1s.
If sorter system actors does not close in time, systeam.closedWg is blocked.

### Check List 

Tests 

 - Unit test

Related changes

 - Need to cherry-pick to the release branch

### Release note 

```release-note
Fix an issue that TiCDC can not exit.
```

### CODE REVIEW

File: cdc/sorter/leveldb/system/system.go [hunk 1]
```diff
@@ -211,37 +197,26 @@ func (s *System) Stop() error {
...
-	s.broadcast(ctx, s.compactRouter, message.StopMessage())
-	// Close metrics goroutine.
-	close(s.closedCh)
-	// Wait actors and metrics goroutine.
-	s.closedWg.Wait()
-
-	// Stop systems.
-	err := s.dbSystem.Stop()
+	// Stop all actors and system to release resource.
+	err := s.WriterSystem.Stop()
```
Comment (3AceShowHand) [inline cdc/sorter/leveldb/system/system.go:201]: for `Stop` related logic, shall all `System` Stop() called, no matter the previous one success or not ?

File: cdc/sorter/leveldb/system/system_test.go [hunk 2]
```diff
@@ -16,7 +16,12 @@ package system
 import (
 	"context"
 	"testing"
+	"time"
 
+	"github.com/pingcap/tiflow/cdc/sorter/leveldb"
+	"github.com/pingcap/tiflow/cdc/sorter/leveldb/message"
+	"github.com/pingcap/tiflow/pkg/actor"
+	actormsg "github.com/pingcap/tiflow/pkg/actor/message"
```
Comment (3AceShowHand) [inline cdc/sorter/leveldb/system/system_test.go:24]: how aboout name it `actorMsg`. `actormsg` would make my linter annoying.

File: cdc/sorter/leveldb/sorter.go [hunk 3]
```diff
@@ -187,14 +187,18 @@ func (ls *Sorter) Run(ctx context.Context) error {
 		err = ctx.Err()
 	case err = <-ls.errCh:
 	}
-	ctx, cancel := context.WithCancel(context.TODO())
-	defer cancel()
 	atomic.StoreInt32(&ls.closed, 1)
+	// We should never lost message, make sure StopMessage is sent.
+	ctxTODO := context.TODO()
```
Comment (3AceShowHand) [inline cdc/sorter/leveldb/sorter.go:192]: IMHO, `ctx := context.TODO()` is preferred. `ctxTODO` is so mouthful.

File: cdc/sorter/leveldb/system/system_test.go (hunk 2 above)
Comment (overvenus) [reply cdc/sorter/leveldb/system/system_test.go:24]: I prefer `actormsg`, like some std package names, `ioutil`, `tzdata`, `expvar`, etc.

File: cdc/sorter/leveldb/system/system.go (hunk 1 above)
Comment (overvenus) [reply cdc/sorter/leveldb/system/system.go:201]: Good point, system.Stop never return error, I'll leave a TODO and fix it in a later PR.

File: cdc/sorter/leveldb/message/task.go [hunk 4]
```diff
@@ -44,6 +45,9 @@ type Task struct {
 	// Deletes all of the key-values in the range.
 	// reader -> leveldb and leveldb -> compactor
 	DeleteReq *DeleteRequest
+
+	// A test message.
+	Test *Test
```
Comment (3AceShowHand) [inline cdc/sorter/leveldb/message/task.go:50]: how about leave a todo message there that `remove test logic later`?

Comment (overvenus) [reply cdc/sorter/leveldb/message/task.go:50]: I don't think they will be removed, I found them are quite handy for testing😅

**Review by sdojjy — APPROVED** [review]

**Review by 3AceShowHand — APPROVED** [review]

**Pull request merged**